source/dub/dub.d

```diff
@@ -391,6 +394,14 @@ class Dub {
 	/// application.
 	@property string projectName() const { return m_project.name; }
 
+	@property string mainRecipePath() const { return m_mainRecipePath; }
+	@property string mainRecipePath(string recipePath) 
+	{ 
+		import std.path:stripExtension, baseName;
+		return m_mainRecipePath = recipePath.baseName.stripExtension;
```

Comment:
why baseName and stripExtension? I would expect this feature to work also in these cases:

```sh
dub --recipe=foo/dub.json
dub --recipe=dub.json # when a dub.sdl also exists
```

Reply:
This one is what is actually used for doing the cache

Reply:
The full path is passed via NativePath(options.recipePath)

Reply:
I have made this step only be executed in the generator.